test/test.jl

```diff
@@ -216,3 +216,15 @@ if has_cuda_gpu()
     @test MultiEvent((event1, event2, event3)) isa Event
   end
 end
+
+  @testset "fallback test: callable types" begin
+    function f end
+    @kernel function (a::typeof(f))(x, ::Val{m}) where m
```

Comment:
Can you add a second test:

```
     @kernel function (a::typeof(f))(x, ::Val{1})
```

to see what happens when two invocations of `@kernel` define the wrappers? You should get a warning from Julia

Reply:
I added a test, everything so far works fine? But I didn't get a warning.